src/bindgen/ir/generic_path.rs

```diff
@@ -10,6 +10,46 @@ use crate::bindgen::ir::{ConstExpr, Path, Type};
 use crate::bindgen::utilities::IterHelpers;
 use crate::bindgen::writer::{Source, SourceWriter};
 
+// Struct that serves as key for resolving associated types
+#[derive(Debug, Clone, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub struct AssocTypeId{
+    pub ty: Box<Type>,
+    pub trait_: Path,
+    pub ident: Path,
+}
+
+impl AssocTypeId{
+    
+    pub fn new(path: &syn::Path, qself: &Option<syn::QSelf>) -> Result<Self, String>{
+        
+        let self_type = if let Some(qself) = qself{
+            &qself.ty
+        } else {
+            return Err(String::from("Type is not fully-qualified."))
+        };
+
+        // Extract type
+        let ir_type_opt = Type::load(&self_type)?;
```

Comment:
what happens if type itself is associated type. Like here:
```rs
trait Kita {
    type Assoc;
}

impl Kita for i32 {
    type Assoc = i32;
}

impl Kita for u32 {
    type Assoc = <i32 as Kita>::Assoc;
}
```

in this case you have to first resolve the inner mapping and then look into the map to get the type. This will be a bit harder to implement, you would have to topologically sort trait impls. Hopefully, we won't need this

Reply:
but if you think you can make this work, then do so